### PR TITLE
Some infrastructure fixes for the repo

### PR DESCRIPTION
**Pull request by mavasani**
1. Always build and deploy VSIXes with cibuild.cmd
2. Always generate TestResults.xml after running unit tests with BuildAndTest.proj
3. Use Microsoft.VSSDK.BuildTools NuGet package instead of relying on the VSSDK installation on the box.
4. Bump the package and assembly version for analyzer packages to 2.0.0 and beta1.
5. Fix BuildNumber/Version computation and add switch to cibuild to trigger official builds.

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @jinujoseph @dotnet/analyzer-ioperation for review

File: build/Targets/Analyzers.Imports.targets [hunk 1]
```diff
@@ -5,29 +5,60 @@
...
-      <ToolsSemanticVersion>0.0.0</ToolsSemanticVersion>
-      <ToolsPreReleaseVersion>beta3</ToolsPreReleaseVersion>
-      <AssemblyVersion>$(SemanticVersion).0</AssemblyVersion>
+    <Version>2.0.0</Version>
+    <RoslynSemanticVersion>$(Version)</RoslynSemanticVersion>
+    <PreReleaseVersion Condition="'$(PreReleaseVersion)' == ''">beta1</PreReleaseVersion>
+    <ToolsSemanticVersion>0.0.0</ToolsSemanticVersion>
+    <ToolsPreReleaseVersion>beta1</ToolsPreReleaseVersion>
+
+    <BuildNumber Condition="'$(BuildNumber)' == ''AND ('$(BUILD_BUILDNUMBER)' != '')">$(BUILD_BUILDNUMBER)</BuildNumber>
```
Comment (mavasani) [inline build/Targets/Analyzers.Imports.targets]: This logic is ported from roslyn repo.

File: build/Targets/References.Targets [hunk 2]
```diff
@@ -17,8 +18,20 @@
         <IncludeAnalyzerReferences>false</IncludeAnalyzerReferences>
         <IncludeToolsetCompilerReference>true</IncludeToolsetCompilerReference>
         <IncludeFakeSignReference>true</IncludeFakeSignReference>
-        <IncludeTestReferences>true</IncludeTestReferences>      
-    </PropertyGroup>
+        <IncludeTestReferences>true</IncludeTestReferences>
```
Comment (jinujoseph) [inline build/Targets/References.Targets]: why are we including TestReferences here ?

**Comment by jinujoseph** [comment]
Tagging @tmat for Info on the infrastructure needs of this repo

**Comment by mavasani** [comment]
> Should we be defaulting OfficialBuild in our project files?

Probably, but I followed our other repo conventions, where it is not defaulted.

**Comment by tmat** [comment]
@mavasani @jinujoseph  Most of these file can be deleted once we start using RepoToolset.

**Comment by JarLob** [comment]
@mavasani, So there will be no non-beta 1.2 release :( ?

**Comment by mavasani** [comment]
Yes, we cannot have one till IOperation is supported. So, even 2.0.0 will have only beta releases.

**Pull request merged**